### PR TITLE
Enhancement/clean up eslint errors

### PR DESCRIPTION
**Pull request by jpica11**
Issue:
ESLint errors that could not be fixed by --fix flag

## What I did
Clean up the obvious issues and add ignore flags for eslint issues that might not be fixable.

## How to test
Run yarn build and demo to ensure all is working as expected

- Does this need an update to the documentation?
  - Not yet, will be needed when husky is added (fast follow to this pr)

### CODE REVIEW

File: app/aem/src/client/preview/helpers/GenericModel.js [hunk 1]
```diff
@@ -5,7 +5,6 @@
 export class GenericModel {
   constructor(content = {}) {
     this.content = content;
-    const self = this;
```
Comment (CodeByAlex) [inline app/aem/src/client/preview/helpers/GenericModel.js]: Nice

File: app/aem/src/client/preview/render.ts [hunk 2]
```diff
@@ -104,12 +113,9 @@ const resetRoot = () => {
...
   const components: any[] = aemMetadata ? aemMetadata.components : [];
-  const info = resourceType ? new ComponentLoader().resolve(resourceType, components) : null;
-  return !template && info
-    ? info.module
-      ? info.module
-      : `unable to load ${resourceType}`
-    : template;
+  let info = resourceType ? new ComponentLoader().resolve(resourceType, components) : null;
+  info = info && info.module ? info.module : `unable to load ${resourceType}`;
+  return !template ? info : template;
```
Comment (CodeByAlex) [inline app/aem/src/client/preview/render.ts]: I like this - it’s easier to read

File: app/aem/src/client/preview/helpers/ResourceResolver.ts [hunk 3]
```diff
@@ -14,21 +15,19 @@ export default class ResourceResolver {
   }
 
   createResourceLoader(parentPath) {
-    if (!Array.isArray(parentPath)) {
-      parentPath = parentPath.split('/').filter(s => !!s);
-    }
+    const parsedParentPath = parentPath.split('/').filter((s: any) => s);
+
```
[the remark this answers is not in the document]
Comment (jpica11) [reply app/aem/src/client/preview/helpers/ResourceResolver.ts]: Actually meant to add strict typing to 'parentPath'. It should always come through as a string, so I don't think the array check is necessary.

File: app/aem/src/client/preview/helpers/GenericModel.js (hunk 1 above)
Comment (jpica11) [reply app/aem/src/client/preview/helpers/GenericModel.js]: 😄

**Pull request merged**